### PR TITLE
Add debug messaging to usage plugin

### PR DESCRIPTION
**Pull request by existentialism**
Followup to https://github.com/babel/babel-preset-env/pull/280

```txt
Polyfills
=========

Added the following polyfills:
  [src/in.js] es6.promise {"ie":"11"}
  [src/in.js] es6.map {"firefox":"50","ie":"11"}
  [src/in.js] regenerator-runtime/runtime {"chrome":"52","firefox":"50","ie":"11"}
  [src/in.js] web.dom.iterable {"chrome":"52","firefox":"50","ie":"11"}
```

### CODE REVIEW

File: src/use-built-ins-plugin.js [hunk 1]
```diff
@@ -26,7 +26,7 @@ function getObjectString(node) {
 export default function({ types: t }) {
   function addImport(path, builtIn, builtIns) {
     if (builtIn && !builtIns.has(builtIn)) {
-      builtIns.add(builtIn);
+      builtIns.set(builtIn, path.hub.file.opts.filename);
```
Comment (existentialism) [inline src/use-built-ins-plugin.js]: Probably need to add some guards here?

File: test/debug-fixtures/usage/stdout.txt [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  transform-regenerator {"firefox":"50","ie":"11"}
+  transform-exponentiation-operator {"firefox":"50","ie":"11"}
+  transform-async-to-generator {"chrome":"52","firefox":"50","ie":"11"}
+  syntax-trailing-function-commas {"chrome":"52","firefox":"50","ie":"11"}
+
+Polyfills
+=========
+
+Added the following polyfills:
+  [src/in.js] es6.promise {"ie":"11"}
```
Comment (hzoo) [inline test/debug-fixtures/usage/stdout.txt]: Would be cool to add the line/col or even the code frame for this (mostly for ourselves to find optimizations)

File: src/use-built-ins-plugin.js (hunk 1 above)
Comment (hzoo) [inline src/use-built-ins-plugin.js]: it's "unknown" in astexplorer, dono if we need a guard

File: src/use-built-ins-plugin.js [hunk 3]
```diff
@@ -274,9 +276,29 @@ Please remove the "require('babel-polyfill')" call or use "useBuiltIns: 'entry'"
...
     pre() {
-      this.builtIns = new Set();
+      this.builtIns = new Map();
       this.usesRegenerator = false;
     },
+    post() {
+      const { debug, onDebug } = this.opts;
+
+      if (debug) {
+        if (!this.builtIns.size) {
```
Comment (hzoo) [inline src/use-built-ins-plugin.js]: Actually what's this?

Comment (hzoo) [inline src/use-built-ins-plugin.js]: oh lol it's map.size.. nvm 😅

**Comment by hzoo** [comment]
Might have a lot of spam from this, but maybe that's ok?

**Comment by existentialism** [comment]
I think it's okay for the spam, since they've opted into `debug` mode.

We could also group by file:

```txt
Added the following polyfills:

  [src/in.js]
    es6.promise {"ie":"11"}
    es6.map {"firefox":"50","ie":"11"}
    regenerator-runtime/runtime {"chrome":"52","firefox":"50","ie":"11"}
    web.dom.iterable {"chrome":"52","firefox":"50","ie":"11"}

  [src/in-2.js]
    es6.promise {"ie":"11"}
    web.dom.iterable {"chrome":"52","firefox":"50","ie":"11"}
```

But yeah, on a big app, it'll produce a ton.

**Pull request merged**